tests/intl/intltest.cpp

```diff
@@ -287,6 +289,60 @@ TEST_CASE("wxTranslations::AddCatalog", "[translations]")
     }
 }
 
+TEST_CASE("wxTranslations::GetBestTranslation", "[translations]")
+{
+    wxFileTranslationsLoader::AddCatalogLookupPathPrefix("./intl");
+
+    const wxString domain("internat");
+
+    wxTranslations trans;
+
+    SECTION("ChooseLanguage")
+    {
+        // Simple case.
+        wxSetEnv("WX_LANGUAGE", "fr:en");
+        CHECK( trans.GetBestTranslation(domain) == "fr" );
+        CHECK( trans.GetBestAvailableTranslation(domain) == "fr" );
+
+        // Choose 2nd language _and_ its base form.
+        wxSetEnv("WX_LANGUAGE", "cs:fr_CA:en");
+        CHECK( trans.GetBestTranslation(domain) == "fr" );
+        CHECK( trans.GetBestAvailableTranslation(domain) == "fr" );
+    }
+
+    SECTION("EnglishHandling")
+    {
+        // Check that existing en_GB file isn't used for msgid language.
+        wxSetEnv("WX_LANGUAGE", "en_US");
+
+        CHECK( trans.GetBestTranslation(domain) == "en" );
+        // GetBestAvailableTranslation() will wrongly return "en_GB", don't test that.
+
+        wxSetEnv("WX_LANGUAGE", "es:en");
+        CHECK( trans.GetBestTranslation(domain) == "en" );
+        // GetBestAvailableTranslation() will wrongly return "en_GB", don't test that.
+
+        // And that it is used when it should be
+        wxSetEnv("WX_LANGUAGE", "en_GB");
+        CHECK( trans.GetBestTranslation(domain) == "en_GB" );
+        CHECK( trans.GetBestAvailableTranslation(domain) == "en_GB" );
+    }
+
+    SECTION("DontSkipMsgidLanguage")
+    {
+        // Check that msgid language will be used if it's the best match.
+        wxSetEnv("WX_LANGUAGE", "cs:en:fr");
+        CHECK( trans.GetBestTranslation(domain) == "en" );
+
+        // ...But won't be used if there's a suitable translation file.
+        wxSetEnv("WX_LANGUAGE", "fr:en:cs");
+        CHECK( trans.GetBestTranslation(domain) == "fr" );
+        CHECK( trans.GetBestAvailableTranslation(domain) == "fr" );
+    }
+
+    wxUnsetEnv("WX_LANGUAGE");
```

Comment:
I'd use `wxON_BLOCK_EXIT` to do this to ensure this variable doesn't remain set and then affect some subsequent test in very hard to find way later.

Reply:
What am I missing here? This seems to be the canonical way to do teardown; the code is executed when a check fails and even if an unexpected exception is thrown anywhere within the SECTION...

Reply:
Right now it's indeed fine, but if anybody adds a `REQUIRE` test, it would exit without executing this line, so it just seems safer to ensure it does always get executed.

Reply:
I see, thanks!